autogpt/memory/message_history.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import annotations
+
+import copy
+import json
+from typing import TYPE_CHECKING
+
+from autogpt.memory.vector.memory_item import MemoryItem
+
+if TYPE_CHECKING:
+    from autogpt.agent import Agent
+
+from autogpt.config import Config
+from autogpt.json_utils.utilities import (
+    LLM_DEFAULT_RESPONSE_FORMAT,
+    is_string_valid_json,
+)
+from autogpt.llm.base import Message
+from autogpt.llm.utils import create_chat_completion
+from autogpt.log_cycle.log_cycle import PROMPT_SUMMARY_FILE_NAME, SUMMARY_FILE_NAME
+from autogpt.logs import logger
+from autogpt.memory.vector import VectorMemory
+
+
+class MessageHistory(list[Message]):
+    agent: Agent
+
+    summary: str
+
+    last_trimmed_index: int
+
+    def __init__(self, agent: Agent):
+        self.agent = agent
```

Comment:
I agree it's not great, but I'm not sure how else to arrange this while still encapsulating the same functionality in `MessageHistory`, because I feel like the set of methods in there makes sense.

Reply:
Yeah, I see. It's the log_cycle bit around the language model.  That's definitely out of scope for you to address.

Ignore for now